haskell/private/actions/link.bzl

```diff
@@ -43,6 +43,10 @@ def _fix_linker_paths(hs, inp, out, external_libraries):
             [
                 "cp {} {}".format(inp.path, out.path),
                 "chmod +w {}".format(out.path),
+                "/usr/bin/install_name_tool -id @rpath/{} {}".format(
```

Comment:
According to the [manpage](https://www.manpagez.com/man/1/install_name_tool/) `-id` changes the “library identification name” of a dynamic shared library. I can’t find out what that is, Darwin’s [ld manpage](https://www.manpagez.com/man/1/ld/) doesn’t mention the word `identification` (or `id` for that matter).

Could you add an inline-comment what this line achieves? Would be awesome to have one for `-change` below as well.

Reply:
I've added some comments explaining this. The linked Oracle blog post also explains this to some extent.